src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContact.hpp

```diff
@@ -155,10 +173,37 @@ class SolidMechanicsLagrangeContact : public ContactSolverBase
 
   real64 m_initialResidual[3] = {0.0, 0.0, 0.0};
 
+  static const localIndex m_maxFaceNodes; // Maximum number of nodes on a contact face
+
   void createPreconditioner( DomainPartition const & domain );
 
   void computeFaceDisplacementJump( DomainPartition & domain );
 
+  template< localIndex DIMENSION, typename POINT_COORDS_TYPE >
+  GEOS_HOST_DEVICE
+  inline static real64 computeDiameter( POINT_COORDS_TYPE points,
```

Comment:
I would suggest moving this method (and the similar one in `src/coreComponents/finiteElement/elementFormulations/ConformingVirtualElementOrder1.hpp`) to the computational geometry source file.